functions/venue.js

```diff
@@ -711,6 +711,35 @@ exports.deleteVenue = functions.https.onCall(async (data, context) => {
   admin.firestore().collection("venues").doc(venueId).delete();
 });
 
+exports.deletePollInVenue = functions.https.onCall(
+  async ({ venueId, pollId }, context) => {
+    checkAuth(context);
+    await checkUserIsAdminOrOwner(venueId, context.auth.token.user_id);
+
+    admin
+      .firestore()
+      .collection("venues")
+      .doc(venueId)
+      .collection("chats")
+      .doc(pollId)
+      .update({ deleted: true });
+  }
+);
+
+exports.voteInPoll = functions.https.onCall(
+  async ({ venueId, pollId, votes }, context) => {
+    checkAuth(context);
+
+    admin
+      .firestore()
+      .collection("venues")
+      .doc(venueId)
+      .collection("chats")
+      .doc(pollId)
+      .update({ votes });
+  }
+);
```

Comment:
You should also store the array of ids of people who voted to avoid double voting

Reply:
The array of `userId`'s that have voted should be what is stored here IMO, we shouldn't keep an arbitrary integer of the count of votes here at all.